miso/index.js

```diff
@@ -0,0 +1,35 @@
+import { WASI, OpenFile, File, ConsoleStdout } from "https://cdn.jsdelivr.net/npm/@bjorn3/browser_wasi_shim@0.3.0/dist/index.js";
+import ghc_wasm_jsffi from "./ghc_wasm_jsffi.js";
+
+// Runs Miso
+async function initialize() {
+  const args = [];
+  const env = ["GHCRTS=-H64m"];
+  const fds = [
+    new OpenFile(new File([])), // stdin
+    ConsoleStdout.lineBuffered((msg) => console.log(`[WASI stdout] ${msg}`)),
+    ConsoleStdout.lineBuffered((msg) => console.warn(`[WASI stderr] ${msg}`)),
+  ];
+  const options = { debug: false };
+  const wasi = new WASI(args, env, fds, options);
+
+  const instance_exports = {};
+  const { instance } = await WebAssembly.instantiateStreaming(fetch("miso/bin.wasm"), {
+    wasi_snapshot_preview1: wasi.wasiImport,
+    ghc_wasm_jsffi: ghc_wasm_jsffi(instance_exports),
+  });
+  Object.assign(instance_exports, instance.exports);
+
+  wasi.initialize(instance);
+  await instance.exports.hs_start();
+}
+
+// Initializes Miso on page load
+window.addEventListener("load", (event) => {
+  initialize();
+});
+
+// Initializes Miso on Turbolinks transition
+document.addEventListener("turbolinks:load", (e) => {
+  initialize();
+});
```

Comment:
This file is essentially a fusion of the JS wrappers from [IHP's Elm boilerplate branch](https://github.com/digitallyinduced/ihp-boilerplate/blob/53756d9de1a3617a7540ab5bed2e71e3673e589c/elm/index.js) and [Tweag's Miso Wasm examples](https://github.com/tweag/ghc-wasm-miso-examples/blob/222aa700cf2e608e4f83a397146391ccb64e8823/frontend/index.js).